src/code-considerations/criteria-for-inclusion.md

```diff
@@ -0,0 +1,70 @@
+# Criteria for inclusion in `std`
+
+**Note**: *The process for deciding what does and does not get added to the standard
+library is too complex to reduce to a set of concrete rules. Ultimately we rely
+upon the expertise of members of the library API team, but here we attempt to
+document the reasoning that has been used historically to guide contributors
+and team members and to ensure long term consistency in how these decisions are
+made even as the team membership naturally changes over time.*
+
+## Reasons For Inclusion
+
+* APIs where there is a clear best implementation
+* APIs where there is a strong benefit to having only one, such as common
+  traits (e.g. `Error`, the `Fail` trait from `failure` crate showed the cost
+  of competing error traits and how it causes ecosystem fractures)
+* Core value propositions: APIs that we want to use in examples for our
+  learning materials to demonstrate rust's core value propositions. (e.g.
+  scoped threads simplifying examples demonstrating thread and memory safety)
```

Comment:
We don't have an argument parsing library, terminal prompting (there have been proposals with somewhat mixed reception I think), regular expressions, http, json or other stuff that would sure be _convenient_ in toy programs.
We don't include a repl either.

The stuff for examples allows the example code to take inputs and make outputs instead of existing in a never-called vacuum.

Perhaps if you mean "hasn't even had a high school programming course before and wants to explore CS101 basics" then it might work. If "getting started" means translating a 20-lines-of-python problem to rust then it doesn't work.

Reply:
I'm definitely thinking more of 'hello world' than a practical program, or like the examples from the Book or similar tutorials

Reply:
I think this is already covered by "want to use in examples" then? Just to have those examples print something we need IO. And "at least as powerful as brainfuck" is table stakes anyway.

`Write`, `stdout()` and `Display` are covered by 

> APIs where there is a strong benefit to having only one

So most of that already falls out of the given reasons.

Reply:
I think it's ok if there is some overlap here? Like, I certainly think it is a reasonable objective that a new programmer can sit down and write a few first programs without needing deps. We can certainly argue about what that means (I would like terminal prompting/basic parsing to be in std, for example, and I don't think that extends to regex/http/json, but I think that it is reasonable to have different opinions here).

Reply:
It still seems pretty very similar to writing documentation. E.g. if hypothetically `println!()` didn't exist and the rust book had to copy-paste some boilerplate to write to stdout in every example then they could open an issue and say hey, this would simplify our tutorial a lot if this function existed.